crates/router/src/types/storage/revenue_recovery.rs

```diff
@@ -67,14 +67,14 @@ impl RevenueRecoveryPaymentData {
 
 #[derive(Debug, serde::Deserialize, Clone, Default)]
 pub struct RevenueRecoverySettings {
```

Comment:
Same concern as I'd pointed out before, because of the `Default` derive, the value for `monitoring_threshold_in_seconds` field would be `0` if it none of the fields in this struct are provided. Are you okay with that?

Also, why is the `revenue_recovery` field in the `Settings` struct required (when the `v2` feature flag is enabled)? Please make that field optional, there are cases where we're deploying v2 builds internally but don't need any revenue recovery features to be enabled...

And consider moving this to the `settings.rs` file maybe?